include/ccf/service/node_info.h

```diff
@@ -69,6 +69,13 @@ namespace ccf
      * node identity in `public_key` field. Service-endorsed certificate is
      * recorded in "public:ccf.nodes.endorsed_certificates" table */
     std::optional<crypto::Pem> cert = std::nullopt;
+
+    /** Commit state for Retired state
+     *
+     * Introduced during 2.x (2.0.5), and so optional for backward
+     * compatibility.
+     */
+    bool retired_committed = false;
```

Comment:
We should also add an assertion in `ledger.py` to make sure no signatures are recorded for a node where this is set.

Reply:
We should really enforce a single signature after `RETIRED`, and this is weaker than that, but it doesn't hurt to check in addition.

Reply:
Yes, the assertion is currently too loose [there](https://github.com/microsoft/CCF/blob/0943d8c4a23985bc1fafd50032594c6f5c1405a1/python/ccf/ledger.py#L521-L528).